src/router_sc_worker.erl

```diff
@@ -326,13 +318,23 @@ get_nonce(PubkeyBin, Ledger) ->
 
 -spec active_sc_expiration() -> {error, no_active_sc} | {ok, pos_integer()}.
 active_sc_expiration() ->
-    case blockchain_state_channels_server:active_sc_id() of
-        undefined ->
+    case blockchain_state_channels_server:active_sc_ids() of
+        [] ->
             {error, no_active_sc};
-        ActiveSCID ->
+        ActiveSCIDs ->
             SCs = blockchain_state_channels_server:state_channels(),
-            {ActiveSC, _} = maps:get(ActiveSCID, SCs),
-            {ok, blockchain_state_channel_v1:expire_at_block(ActiveSC)}
+            [SoonestSCIDToExpire | _] =
+                lists:sort(
+                    fun(SCIDA, SCIDB) ->
+                        {ActiveSCA, _} = maps:get(SCIDA, SCs),
+                        {ActiveSCB, _} = maps:get(SCIDB, SCs),
+                        blockchain_state_channel_v1:expire_at_block(ActiveSCA) <
+                            blockchain_state_channel_v1:expire_at_block(ActiveSCB)
+                    end,
+                    ActiveSCIDs
+                ),
+            {SoonestSCToExpire, _} = maps:get(SoonestSCIDToExpire, SCs),
```

Comment:
This will blow up if `SoonestSCIDToExpire` is not found in `SCs`, presumably that's the behavior you want here?

Reply:
yeah this would be very bad if we had a active SC ID but not the actual SC, this is why I always use `maps:get/2`